great_expectations/checkpoint/actions.py

```diff
@@ -328,13 +352,17 @@ def _run(  # type: ignore[override] # signature does not match parent  # noqa: C
             # To send a notification with a link to the validation result, we need to have created the validation  # noqa: E501
             # result in cloud. If the user has configured the store action after the notification action, they will  # noqa: E501
             # get a warning that no link will be provided. See the __init__ method for ActionListValidationOperator.  # noqa: E501
-            if (
-                "store_validation_result" in payload
-                and "validation_result_url" in payload["store_validation_result"]
-            ):
-                validation_result_urls.append(
-                    payload["store_validation_result"]["validation_result_url"]
-                )
+            action_context = action_context or ActionContext()
+            store_validation_results = action_context.filter_results(
+                class_=StoreValidationResultAction
+            )
+            if store_validation_results:
```

Comment:
This line leads me to think there's a special case other than empty list that's being accounted for, but I don't think that's actually the case. Why not remove the top-level `if`, and just let the `for` handle empty lists?

Reply:
Oh yeah that should be fine. I'll do that now